File: asignaturas/sql/Asignaturas-IvanRonco/Profesores_Y_Modulos_BD-Java/src/main/java/dto/profesor/ProfesorDto.java
package dto.profesor;

import com.squareup.moshi.Json;
import dto.modulo.ListaModulosDto;
import dto.modulo.ModuloDto;
import lombok.AllArgsConstructor;
import lombok.Data;
import org.simpleframework.xml.Element;
import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Root;

import java.util.List;

@Data
@AllArgsConstructor
@Json(name = "profesor")
@Root(name = "profesor")
public class ProfesorDto {
    @Json(name = "profesor_id")
    @Element(name = "profesor_id")
    String id;

    @Json(name = "nombre")
    @Element(name = "nombre")
    String nombre;

    @Json(name = "fecha_incorporación")
    @Element(name = "fecha_incorporación")
    String fechaIncorporacion;

    @Json(name = "modulos")
    @ElementList(name = "modulos")
    List<ModuloDto> modulosDto;
}

File: asignaturas/sql/SergioDeIscar-Asignatura-Kotlin/SergioDeIscar-Asignatura-Kotlin/src/main/kotlin/dto/ProfesorDto.kt
package dto

import org.simpleframework.xml.Attribute
import org.simpleframework.xml.Element
import org.simpleframework.xml.Root

@Root(name = "profesor")
data class ProfesorDto(
    @field:Attribute(name = "id")
    @param:Attribute(name = "id")
    val id: String = "0",

    @field:Element(name = "nombre")
    @param:Element(name = "nombre")
    val nombre: String,

    @field:Element(name = "incorporacion")
    @param:Element(name = "incorporacion")
    val fechaIncorporacion: String,

    @field:Element(name = "modulo")
    @param:Element(name = "modulo")
    val modulos: ModulosDto
)